stdatomic.h: No such file or directory on GCC 4.8 due to false positive C11 atomics detection in os_core_unix.c

### Describe the bug

fatal error: stdatomic.h: No such file or directory when building pjproject 2.16 on GCC 4.8. Commit #4570 ("Use C11 atomics when available") introduced a false positive in the C11 atomics detection logic. When building with -std=gnu11, __STDC_VERSION__ >= 201112L evaluates to true on GCC 4.8, causing the build to unconditionally include stdatomic.h — which does not exist in GCC 4.8 (support was added in GCC 4.9).

### Steps to reproduce

Use a system with GCC 4.8 (CentOS 7 / RHEL 7)
Attempt to build pjproject 2.16
Build fails with fatal error: stdatomic.h: No such file or directory

### PJSIP version

2.16

### Context

OS: CentOS 7
GCC: 4.8.5 (Red Hat 4.8.5-44)
pjproject: 2.16
Build flag in use: -std=gnu11
Introduced by: commit #4570 "Use C11 atomics when available"``

### Log, call stack, etc

```shell
../src/pj/os_core_unix.c:52:27: fatal error: stdatomic.h: No such file or directory
 #    include <stdatomic.h>
```

--- a/pjlib/src/pj/os_core_unix.c
+++ b/pjlib/src/pj/os_core_unix.c
@@ -47,7 +47,10 @@
 
 #if PJ_HAS_THREADS
 #  if defined(__STDC_VERSION__) && __STDC_VERSION__ >= 201112L \
-                                && !defined(__STDC_NO_ATOMICS__)
+                                && !defined(__STDC_NO_ATOMICS__)  \
+                                && (!defined(__GNUC__) || defined(__clang__) \
+                                || __GNUC__ > 4 \
+                                || (__GNUC__ == 4 && __GNUC_MINOR__ >= 9))
 #    define HAS_STD_ATOMICS 1
 #    include <stdatomic.h>
 #  else

